src/components/AppShell/AppShell.jsx

```diff
@@ -27,16 +30,37 @@ const AppShell = ({
   displaySkipLink,
 }) => {
 
-  const enabledProducts = ['publish', 'analyze'];
-  if (featureFlips.includes('enableReply')) {
-    enabledProducts.push('reply');
-  }
+  const engageEnabled = enabledProducts.includes('engage');
+
+  const products = [
+    {
+      id: 'publish',
+      isNew: false,
+      visible: true,
+      enabled: true,
+      enableURL: null
```

Comment:
I think I might switch this to be `href` and either have the signup URL for the product or the actual product URL.

Right now we're passing all these attributes as a structure (giving them a meaning here), and then some layers down below we need to know also that meaning to build the product URL as well as to filter down the visible products, while we know it all in here.

So it might just look like:

```jsx
const products = ['publish', 'analyze']
if (engageEnabled || featureFlips.includes('engageRollOut')) {
    products.push('engage')
}
products.map(product => {
    if (product === 'engage') {
        return {
            id: product,
            isNew: true,
            href: engageEnabled ? `https://${product}.buffer.com` : ENABLE_ENGAGE_URL
        }
    }
    return {
        id: product,
        isNew: false,
        href: `https://${product}.buffer.com`
    }
})
```

This would remove the filter logic down below as well as having to build the URLs depending on another flag :)

How does that sound to you, @ssvmvss?

Reply:
Ah, yes! Good one @msanroman! 🙌Initially I had the idea of doing the same I did for engage with all products in order to redirect users to the product URL or to the alternative URL (cross selling page or marketing page) but then I decided to not do that and I never came back to here in order to cleanup the code! Great catch! 🌮

Reply:
I was filtering by `visible` later on 🤔I don't remember why but that makes no sense at all 🙈It does not look like it but I swear I reviewed the PR! 😅Thanks @msanroman 🌮